platforms/posix/src/px4/common/tasks.cpp

```diff
@@ -235,7 +228,7 @@ px4_task_t px4_task_spawn_cmd(const char *name, int scheduler, int priority, int
 		}
 	}
 
-	if (i >= PX4_MAX_TASKS) {
+	if (taskid >= PX4_MAX_TASKS) {
```

Comment:
This is not correct: `taskid` won't be set if all tasks are used.

Reply:
Oh, I see the problem. I should probably not bother trying to clean that up. I'm just making it worse and wrong.